### ISSUE TITLE
perf(model): fuse 4 head Linears into one projection (P3)

### ISSUE THREAD

**Issue by madsbolaris** [madsbolaris/MageZero#3]
**Expected speedup:** 1.5-2× MLP forward.

## Problem
[`src/magezero/model.py#L69-L80`](../blob/macos-support/src/magezero/model.py#L69-L80):
```python
self.player_priority_head   = nn.Linear(hidden_dim_mlp, policy_size_A)  # 256→128
self.opponent_priority_head = nn.Linear(hidden_dim_mlp, policy_size_A)  # 256→128
self.target_head            = nn.Linear(hidden_dim_mlp, policy_size_A)  # 256→128
self.binary_head            = nn.Linear(hidden_dim_mlp, 2)              # 256→2
```

Four separate GEMMs against the same 256-d input `h`. Poor cache reuse on CPU.

## Fix
One fused projection, slice the output:
```python
self.fused_heads = nn.Linear(hidden_dim_mlp, 3*policy_size_A + 2 + 1)
...
fused = self.fused_heads(h)
pA   = fused[..., :128]
pB   = fused[..., 128:256]
tgt  = fused[..., 256:384]
bin2 = fused[..., 384:386]
val  = torch.tanh(fused[..., 386:387]).squeeze(-1)
```

## Risk
Low. Architectural identity for inference; for training, weight init differs slightly so a quick eval-loss sanity check is wise. Old checkpoints will not load — bump model version or provide a migration helper.
